cpp/src/prims/kv_store.cuh

```diff
@@ -945,6 +946,30 @@ class kv_store_t {
   {
   }
 
+  /* when use_binary_search = true */
+  template <bool binary_search = use_binary_search>
+  kv_store_t(rmm::device_uvector<key_t>&& keys,
+             decltype(allocate_dataframe_buffer<value_t>(0, rmm::cuda_stream_view{}))&& values,
+             decltype(cugraph::invalid_idx<key_t>::value)
+               invalid_value /* invalid_value is returned when match fails for the given key */,
```

Comment:
This should be `value_t invalid_value`, and `invalid_value` should be a `thrust::tuple_type`. I think you introduced to fix a compile error, could you specify the code causing the error? I think we should fix this in the caller site, not here.

Reply:
https://github.com/rapidsai/cugraph/blob/branch-24.06/cpp/src/utilities/collect_comm.cuh#L302

```
  kv_store_t<vertex_t, value_t, true> kv_map(std::move(unique_int_vertices),
                                             std::move(tmp_value_buffer),
                                             invalid_vertex_id<vertex_t>::value,
                                             false,
                                             handle.get_stream());
```

Is this the culprit? Let me think about how to fix this.

Reply:
You can just provide `value_t{}`. Here, kv_map is guaranteed to have values for all the input keys, so `invalid_value` should never be used.

Reply:
We never tried to update a thrust::tuple object from a scalar. This is just an undetected bug as the old CCCL allowed this behavior.

Reply:
@seunghwak it's possible your changes fixed this, but IIRC the issue was from this callsite: https://github.com/rapidsai/cugraph/blob/562b5a5b9f3db29184390c319468ccb488d21056/cpp/src/prims/per_v_transform_reduce_dst_key_aggregated_outgoing_e.cuh#L753-L759

Reply:
Yes, same here, let me double check but I think `invalid_vertex_id<vertex_t>::value` should be `edge_src_value_t{}`

Reply:
Yes, this caller site code mistakenly passes an invalid key (instead of an invalid value).

`invalid_vertex_id<vertex_t>::value => edge_src_value_t{}` (as invalid values are irrelevant in this context).

Reply:
Thanks, I'll rebuild locally and verify these changes work with the new constructor removed.

Reply:
Thanks for fixing cugraph bugs :-)

Reply:
No problem! [My latest commit](https://github.com/rapidsai/cugraph/pull/4404/commits/06dee7ebae8b87e1198eabf1b7837bbcc1388a3d) builds for me locally w/ CCCL 2.5, so I think we're good to go.